silences-gitops/README.md

```diff
@@ -0,0 +1,114 @@
+---
+creation_date: 2025-02-04
+issues: []
+owners:
+- https://github.com/orgs/giantswarm/teams/team-atlas
+state: review
+summary: Deploy observability silences using GitOps
+---
+
+# Deploy observability silences using GitOps
+
+The Observability Platform makes use of [Silences](https://github.com/giantswarm/silence-operator/#how-does-it-work) CRs in order to mute alerts, see [silences management](https://intranet.giantswarm.io/docs/support-and-ops/processes/silence-management/).
+
+Those Silences CRs are currently stored in a private Github repository at [https://github.com/giantswarm/silences][1], where GiantSwarm employees can create and manage silences. They are deployed to management clusters via a CronJob which pulls Silences CRs from the Github repository.
+
+## Problem statement
+
+The current deployment solution using a CronJob suffers from both privacy and security issues and does not allow customers to manage their silences. It is also not in line with any of the deployment mechanisms we currently have in place.
+
+## Solution proposal
+
+This solution proposes to re-use the current [management-cluster-base][2] (MCB) and customer management cluster (CMC) (e.g. [giantswarm-management-clusters](https://github.com/giantswarm/giantswarm-management-clusters)) GitOps mechanism in place in order to manage and deploy Silences CRs accross management clusters.
+
+This solution is laid out as follows:
+
+- Silence CRs common to all clusters are stored in the MCB repository, under the `bases/silences/` directory
+- Silence CRs added in MCB also need to be referenced in the `bases/silences/kustomization.yaml` file
+- Silence CRs specific to a cluster are stored in the corresponding CMC repository, under the `management-clusters/<management-cluster-name>/silences/` directory
+- Name collision between MCB and CMC Silences CRs is prevented by adding the special `global-` name prefix when deploying Silences CRs from the MCB repository
```

Comment:
Good catch but I definitely do not like the global prefix (no clue what I would use tbh) but we use shared, common and so on in other places so this feels wrong somehow

Reply:
I am open to other prefix name suggestion here

Reply:
There is a feature in `kustomzie` to automatically add `.metadata.name` prefixes and suffixes that you could utilize and "enforce" this: https://github.com/kubernetes-sigs/kustomize/blob/master/examples/transformerconfigs/README.md#prefixsuffix-transformer, for example: https://github.com/giantswarm/konfigure-operator/blob/0d480e9404625b1278e108ae817980405ddb714a/config/default/kustomization.yaml#L9

Reply:
Maybe maybe you could also add a patch to the kustomization to add a label / annotation that it is one from MCB for all silences CRs flowing thru that kustomization. 🤔

Reply:
I am exactly using the `namePrefix` there https://github.com/giantswarm/management-cluster-bases/pull/177/files#diff-0a7c294daac0ed3af0691830ed653c0139a0da4b40254d70c701529a9732a64b

Yes having a patch to label Silences CRs with their source repository would be helpful as mentioned by Zach earlier.

Reply:
Changed the prefix to `common`